### PR TITLE
added persistence to fasthtml app using dict

### PR DESCRIPTION
**Pull request by erik-dunteman**
Persists checkbox state even when function times out from inactivity.

Uses cool asgi app trait: shutdown event handling.

Confirmed shutdown events work in context of Modal, both from CLI keyboard interrupt and normal function timeout.

### Type of Change

- [ ] New example
- [x] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

## Checklist

- [x] Example is testable in synthetic monitoring system, or `lambda-test: false` is added to example frontmatter (`---`)
  - [x] Example is tested by executing with `modal run` or an alternative `cmd` is provided in the example frontmatter (e.g. `cmd: ["modal", "deploy"]`)
  - [x] Example is tested by running with no arguments or the `args` are provided in the example frontmatter (e.g. `args: ["--prompt", "Formula for room temperature superconductor:"]`
- [x] Example is documented with comments throughout, in a [_Literate Programming_](https://en.wikipedia.org/wiki/Literate_programming) style.
- [x] Example does _not_ require third-party dependencies to be installed locally
- [x] Example pins its dependencies
  - [x] Example pins container images to a stable tag, not a dynamic tag like `latest`
  - [x] Example specifies a `python_version` for the base image, if it is used
  - [x] Example pins all dependencies to at least minor version, `~=x.y.z` or `==x.y`
  - [x] Example dependencies with `version < 1` are pinned to patch version, `==0.y.z`

## Outside contributors

You're great! Thanks for your contribution.

### CODE REVIEW

File: 07_web_endpoints/fasthtml_app.py [hunk 1]
```diff
@@ -21,37 +22,17 @@
...
-body {
-  background-color: #1d1d1d;
-}
-
-[type="checkbox"]:is(:checked, :checked:focus) {
-  --pico-border-color: #7fee64;
-  --pico-background-color: #7fee64;
-}
+css_path_local = Path(__file__).parent / "fasthtml_app.css"
+css_path_remote = PurePosixPath("/assets/fasthtml_app.css")
```
Comment (charlesfrye) [inline 07_web_endpoints/fasthtml_app.py:26]: Why not just a `Path` here?

Comment (erik-dunteman) [reply 07_web_endpoints/fasthtml_app.py]: Screenshot 2024-10-02 at 3 16 45 PM

Function signature on from_local_file specifies PurePosixPath for the remote path, for some reason
Screenshot 2024-10-02 at 3 17 01 PM

Comment (erik-dunteman) [reply 07_web_endpoints/fasthtml_app.py]: though just changed to Path for consistency and it was still accepted

**Comment by erik-dunteman** [comment]
@charlesfrye to avoid weird merge issues, just put the basic fastHTML example into this PR

**Comment by erik-dunteman** [comment]
the following should be ready to go
- add styling to checkbox app
- move checkbox app to `fasthtml_checkboxes.py`
- brought original hello world back into `fasthtml_example.py`

**Comment by charlesfrye** [comment]
superceded #908

**Pull request closed without merging**